backend/logic/curriculums/curriculums.js

```diff
@@ -0,0 +1,95 @@
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format");
+
+var createCurriculum = async (name, type, department, numOfElectives, cores, techComps, comps) => {
+    let error = false;
+  
+    if (!format.verifyCurriculumName(name)) {
+      error = "Invalid curriculum name";
+    } else if(!format.verifyCurrType(type)){
+      error = "Invalid curriculum type";
+    } else if(!format.verifyDepartment(department)){
+      error = "Invalid department";
+    } else if(!format.verifyNumOfElectives(numOfElectives)) {
+      error = "Invalid number of electives"
+    }
+  
+    if (!error == false) {
+      console.error(error);
+      throw new Error(error);
+    }
+  
+    format.verifyCourse(cores);
+    format.verifyCourse(techComps);
+    format.verifyCourse(comps);
+  
+    let connection = await mysql.getNewConnection();
+  
+    try {
+      await connection.beginTransaction();
+      await connection.query("INSERT INTO curriculums (curriculum_name, type, department, numOfElectives) VALUES(?, ?, ?, ?);", [
+        name,
+        type,
+        department,
+        numOfElectives
+      ]);
+      for (let core in cores)
+      {
+        let course_count = await connection.query("SELECT COUNT(*) FROM courses WHERE course_code = ?;", [core]);
+  
+        if(course_count){
+        
+          await connection.query("INSERT INTO curriculum_core_class (curriculum_name, course_code) VALUES(?, ?);", [
+            name,
+            core
+          ]);
+  
+        } else {
+          throw new Error("One of these courses does not yet exist.");
+        }
+      }
+      for (let techComp in techComps)
+      {
+        let course_count = await connection.query("SELECT COUNT(*) FROM courses WHERE course_code = ?;", [techComp]);
+  
+        if(course_count){
+        
+          await connection.query("INSERT INTO curriculum_tech_comp (curriculum_name, course_code) VALUES(?, ?);", [
+            name,
+            techComp
+          ]);
+  
+        } else {
+          throw new Error("One of these courses does not yet exist.");
+        }
+      }
+      for (let comp in comps)
+      {
+        let course_count = await connection.query("SELECT COUNT(*) FROM courses WHERE course_code = ?;", [comp]);
+        if(course_count){
+        
+          await connection.query("INSERT INTO curriculum_complementaries (curriculum_name, course_code) VALUES(?, ?);", [
+            name,
+            comp
+          ]);
+  
+        } else {
+          throw new Error("One of these courses does not yet exist.");
+        }
+      }
+      
+      await connection.commit();
+      connection.release();
+      return true;
+    } catch (error) {
+      connection.rollback();
+      connection.release();
+      console.error(error);
+      throw new Error(false);
+    }
```

Comment:
Could you put the connection.release() in a finally block instead? 
```
try{
    ..
}catch{
    ..
} finally{
    conection.release
}
```

Reply:
The thing is you can't really do that, because if the an error is thrown and the program goes into the catch, the connection will never actually be released since an error is thrown. In other words, if it hits the catch, it will never go into the finally.

Reply:
nope, the finally will always run no matter what happened before

Reply:
kk ill fix it then